### PR TITLE
Select component does not set value to selected option Fix

### PR DESCRIPTION
**Pull request by IanKBovard**
#### What does this PR do?
Fixes issue #3851 by adding a condition to FormFields onBlur event. onBlur will not setState if the current node clicked has a role of menubar.

#### Where should the reviewer start?
FormField.js
FormFieldBox onBlur event.

#### What testing has been done on this PR?
Tested in storybook, using my own custom stories.

#### How should this be manually tested?
Storybook. Should wrap every necessary component in FormField to see if focus behaves as it should.

#### Any background context you want to provide?
When the Select component was wrapped in a FormField component it would take two clicks to run onChange. This was because FormField has setState during an onBlur event which caused a rerender. 

#### What are the relevant issues?
#3851 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
If any other component has a role of menubar, it would affect them.

### CODE REVIEW

**Comment by IanKBovard** [comment]
Not sure, but it might affect this issue as well.
https://github.com/grommet/grommet/issues/3854

**Comment by IanKBovard** [comment]
To test i mostly used this section of storybook. Tabbed through and click on everything to see if the components gained and lost focus.
https://storybook.grommet.io/?path=/story/form--all

**Comment by IanKBovard** [comment]
More explanation:

The issue doesn't lie with the value not setting as described in issue #3851. The problem is that when setState is called within an onBlur event the component re-renders. And, more often than not onBlur triggers before onClick so onClick never fires.

https://www.eventbrite.com/engineering/a-story-of-a-react-re-rendering-bug/

What, I believe, is happening is that when you click on an option in a select dropdown onBlur is being called triggering the setFocus(true). This causes a re-render of the dropdown, which is why the scrollbar resets to the top. When you click a dropdown option again onClick fires, because the dropdown already has focus so onBlur doesn't fire again.

The solution here adds a conditional within onBlur to ignore dropdown items.

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -327,7 +327,10 @@ const FormField = forwardRef(
           if (onFocus) onFocus(event);
         }}
         onBlur={event => {
-          setFocus(false);
+          const role = event.target.getAttribute('role');
```
Comment (ericsoderberghp) [inline src/js/components/FormField/FormField.js]: This seems like a fragile approach. I think we should look for an alternate approach to addressing this.

**Comment by ericsoderberghp** [comment]
I'm wondering why the FormField is getting focus when clicking an option, especially why it is tied to a long options list.

File: src/js/components/FormField/FormField.js (hunk 1 above)
Comment (IanKBovard) [reply src/js/components/FormField/FormField.js]: gotcha, can do.

**Comment by IanKBovard** [comment]
The problem isn't FormField gaining focus, it's that FormField loses focus causing the onBlur to trigger. When you setState in onBlur it causes a rerender.

Here's what happens.
 - I click the dropdown in formfield
 - my list comes up
- i scroll through the list
 - i select on option
- formfield loses focus and the select options gain it
- formfield's onblur fires with setstate causing a rerender
- the scrollbar resets to the top of the selection
- i scroll back down to my option and select it again
- onclick fires

**Comment by IanKBovard** [comment]
updated logic. checking for tag name instead. 
```
if (event.target.tagName === 'INPUT')  setFocus(false);
```

**Comment by IanKBovard** [comment]
will come back to this issue when my work on discover has completed

**Pull request closed without merging**